lighthouse-core/config/config-plugin.js

```diff
@@ -141,6 +145,56 @@ class ConfigPlugin {
     };
   }
 
+
+  /**
+   * Extract and validate groups JSON added by the plugin.
+   * @param {unknown} groupsJson
+   * @param {string} pluginName
+   * @return {Record<string, LH.Config.GroupJson>|undefined}
+   */
+  static _parseGroups(groupsJson, pluginName) {
+    if (groupsJson === undefined) {
+      return undefined;
+    }
+
+    if (!isObjectOfUnknownProperties(groupsJson)) {
+      throw new Error(`${pluginName} group json is not valid.`);
+    }
+
+    if (typeof groupsJson !== 'object') {
+      throw new Error(`${pluginName} has an invalid group type.`);
+    }
+    const groupIds = Object.keys(groupsJson);
+    if (!groupIds.length) {
+      throw new Error(`${pluginName} has an an empty groups object.`);
+    }
+    /** @type {Record<string, LH.Config.GroupJson>} **/
+    const parsedGroupsJson = {};
+    groupIds.forEach(groupId => {
```

Comment:
super nit: could just use `Object.entries` since the first thing you do is grab the value